libcudacxx/test/libcudacxx/std/containers/sequences/inplace_vector/comparison.pass.cpp

```diff
@@ -0,0 +1,93 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++11
+
+#include <cuda/std/cassert>
+#include <cuda/std/initializer_list>
+#include <cuda/std/inplace_vector>
+#include <cuda/std/type_traits>
+
+#include "test_iterators.h"
+#include "test_macros.h"
+#include "types.h"
+
+template <class T>
+__host__ __device__ constexpr void test_equality()
+{
+  using inplace_vector = cuda::std::inplace_vector<T, 42ull>;
+
+  inplace_vector lhs{T(1), T(42), T(1337), T(0)};
+  inplace_vector rhs{T(0), T(1), T(2), T(3), T(4)};
+
+  auto res_equality = lhs == lhs;
+  static_assert(cuda::std::is_same<decltype(res_equality), bool>::value, "");
+  assert(res_equality);
+
+  auto res_inequality = lhs != rhs;
+  static_assert(cuda::std::is_same<decltype(res_inequality), bool>::value, "");
+  assert(res_inequality);
+}
+
+template <class T>
+__host__ __device__ constexpr void test_relation()
+{
```

Comment:
Important: the specification contains `operator<=>`, so we should also add a test for it.

Reply:
We do not have proper support for <=> on device, so I cannot add that as easily

Reply:
Then we should at least leave a TODO comment that we should add a test for <=> once device compiler support is major enough.

Reply:
Oh believe me there is going to be all kinds of fun when this ever happens